### PR TITLE
Fix bug in puppetserver.conf.epp template.

### PR DESCRIPTION
**Pull request by pillarsdotnet**
This is what has been kicking my butt all week.

The packaged [`puppetserver.conf`](https://github.com/puppetlabs/puppetserver/blob/master/resources/ext/config/conf.d/puppetserver.conf#L14) file contains a default `gem-path` setting as follows:
```
    # This setting defines the complete "GEM_PATH" for jruby.  If set, it should include
    # the gem-home directory as well as any other directories that gems can be loaded
    # from (including the vendored gems directory for gems that ship with puppetserver)
    gem-path: [${jruby-puppet.gem-home}, "/opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems", "/opt/puppetlabs/puppet/lib/ruby/vendor_gems"]
```

This setting first appeared at tag [2.7.0](https://github.com/puppetlabs/puppetserver/blob/2.7.0/resources/ext/config/conf.d/puppetserver.conf#L14).  The default value most recently changed at tag [5.3.0](https://github.com/puppetlabs/puppetserver/blob/5.3.0/resources/ext/config/conf.d/puppetserver.conf#L14).

However, [pupmod's](https://github.com/simp/pupmod-simp-pupmod) corresponding [`puppetserver.conf.epp`](https://github.com/simp/pupmod-simp-pupmod/blob/master/templates/etc/puppetserver/conf.d/puppetserver.conf.epp#L13) lacks this setting.

Without a proper `gem-path`, the Puppetserver cannot find certain gems, including `hiera-eyaml`, and will fail to run if there is an eyaml lookup in the hierarchy .  To illustrate:

```
[root@puppetserver ~]# /opt/puppetlabs/server/bin/puppetserver gem list hiera-eyaml

*** LOCAL GEMS ***

hiera-eyaml (3.0.0)
[root@puppetserver ~]# cp /etc/puppetlabs/puppetserver/conf.d/puppetserver.conf . ; grep -v gem-path puppetserver.conf > /etc/puppetlabs/puppetserver/conf.d/puppetserver.conf ; /opt/puppetlabs/server/bin/puppetserver gem list hiera-eyaml
cp: overwrite _./puppetserver.conf_?
[root@cusppupent01 ~]# unalias cp
[root@cusppupent01 ~]# /opt/puppetlabs/server/bin/puppetserver gem list hiera-eyaml

*** LOCAL GEMS ***

hiera-eyaml (3.0.0)
[root@puppetserver ~]# cp /etc/puppetlabs/puppetserver/conf.d/puppetserver.conf . ; grep -v gem-path puppetserver.conf > /etc/puppetlabs/puppetserver/conf.d/puppetserver.conf ; /opt/puppetlabs/server/bin/puppetserver gem list hiera-eyaml

*** LOCAL GEMS ***


[root@puppetserver ~]# service puppetserver restart ; puppet agent -t
Redirecting to /bin/systemctl restart puppetserver.service
Info: Using configured environment 'dev'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Retrieving locales
Info: Loading facts
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Function Call, Function Load Error for function 'eyaml_lookup_key': Lookup using eyaml lookup_key function is only supported when the hiera_eyaml library is present (file: /etc/puppetlabs/code/environments/dev/site.pp, line: 1, column: 1) on node cusppupent01.internal.cnngad.com
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
[root@puppetserver ~]# diff -u /etc/puppetlabs/puppetserver/conf.d/puppetserver.conf puppetserver.conf
--- /etc/puppetlabs/puppetserver/conf.d/puppetserver.conf       2019-05-17 14:02:43.332542701 +0000
+++ puppetserver.conf   2019-05-17 14:02:43.329542639 +0000
@@ -14,6 +14,7 @@
     # This setting defines the complete "GEM_PATH" for jruby.  If set, it should include
     # the gem-home directory as well as any other directories that gems can be loaded
     # from (including the vendored gems directory for gems that ship with puppetserver)
+    gem-path: [${jruby-puppet.gem-home}, "/opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems", "/opt/puppetlabs/puppet/lib/ruby/vendor_gems"]

     # PLEASE NOTE: Use caution when modifying the below settings. Modifying
     # these settings will change the value of the corresponding Puppet settings
[root@puppetserver ~]# cat puppetserver.conf > /etc/puppetlabs/puppetserver/conf.d/puppetserver.conf ; service puppetserver restart ; puppet agent -t
Redirecting to /bin/systemctl restart puppetserver.service
Info: Using configured environment 'dev'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Retrieving locales
Info: Loading facts
Info: Caching catalog for cusppupent01.internal.cnngad.com
Info: Applying configuration version '1558101229'
Notice: System Reboot Required Because:
  audit => modified
Notice: Applied catalog in 22.05 seconds
[root@puppetserver ~]# /opt/puppetlabs/server/bin/puppetserver gem list hiera-eyaml

*** LOCAL GEMS ***

hiera-eyaml (3.0.0)
[root@puppetserver ~]#
```

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,7 +1,6 @@
 * Fri May 17 2019 Robert Vincent <robert.vincent@conning.com> - 7.10.1-0
 - Add missing gem-path setting to puppetserver.conf templae.
```
Comment (pillarsdotnet) [inline CHANGELOG]: Typo: replace `templae` with `template`.

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-pupmod/pipelines/61879868

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-pupmod/pipelines/61883432

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-pupmod/pipelines/61901278

**Comment by trevor-vaughan** [comment]
OEL Acceptance Test is failing, attempting to debug.

**Comment by trevor-vaughan** [comment]
Can't make it fail locally. Cleared the runner caches and trying again.

**Pull request merged**